File: loquacious/src/main/java/com/saantiaguilera/loquacious/observer/LocaleSubscribable.java
package com.saantiaguilera.loquacious.observer;

import android.support.annotation.NonNull;

/**
 * Created by saguilera on 11/18/17.
 */

public interface LocaleSubscribable {
    boolean subscribe(@NonNull OnLocaleChanged subscriptor);
    boolean unsubscribe(@NonNull OnLocaleChanged subscriptor);
}

File: loquacious/src/main/java/com/saantiaguilera/loquacious/observer/LocaleSubscribable.kt
package com.saantiaguilera.loquacious.observer

import java.util.*

/**
 * Created by saguilera on 11/18/17.
 */
interface LocaleSubscribable {
    fun subscribe(subscriptor: OnLocaleChanged): Boolean
    fun unsubscribe(subscriptor: OnLocaleChanged): Boolean
}

typealias OnLocaleChanged = (Locale) -> Unit